plugin/arcconf/arcconf.py

```diff
@@ -0,0 +1,963 @@
+        disk_path = ''
+        if disk_path:
+            vpd83 = LocalDisk.vpd83_get(disk_path)
+        else:
+            vpd83 = ''
+        rpm = arcconf_disk['rotationalSpeed']
+        rpm = 7200
+        link_type = _disk_link_type_of(arcconf_disk)
+
+
+        return Disk(
+            disk_id, disk_name, disk_type, blk_size, blk_count,
+            status, sys_id, _plugin_data=plugin_data, _vpd83=vpd83,
+            _location=disk_location, _rpm=rpm, _link_type=link_type)
+
+    @_handle_errors
+    def disks(self, search_key=None, search_value=None,
+              flags=Client.FLAG_RSVD):
+
+        rc_lsm_disks = []
+        controllerList = []
+        controllerInfo = {'Status':'Optimal', 'Slot':'255', 'Mode':'HBA', 'Name':'Adaptec', 'Serial Number':'1234'}
+
+        getconfig_disks_info = []
+        total_cntrl = 0
+        test = self._arcconf_exec(['list' ], COMPONENT[5], False)
+        output_lines = [
+        l for l in test.split("\n")]
+
+        sys_id = ''
+        for line_num in range(len(output_lines)):
+            cur_line = output_lines[line_num]
+            tmp = cur_line.split(":")
+            if tmp[0].strip() == 'Controllers found':
+                total_cntrl = int(tmp[1].strip())
+
+        for cntrl in range(total_cntrl):
+            for line_num in range(len(output_lines)):
+                cur_line = output_lines[line_num]
+                tmp = cur_line.split(":")
+                cntrl_num = 'Controller %d' %(cntrl + 1)
+                if tmp[0].strip() == cntrl_num :
+                    cntrl_info = tmp[2].split(",")
+                    controllerInfo['Status'] = cntrl_info[0].strip()
+                    controllerInfo['Slot'] = cntrl_info[1].strip()
+                    controllerInfo['Mode'] = cntrl_info[2].strip()
+                    controllerInfo['Name'] = cntrl_info[3].strip()
+                    controllerInfo['Serial Number'] = cntrl_info[4].strip()
+                    sys_id = controllerInfo['Serial Number']
+            cntrl_num = int(cntrl + 1)
+            disks_info = self._arcconf_exec([ "getconfigJSON", str(cntrl+1)], COMPONENT[5], False)
+            disk_info_json = disks_info.split("\n")[1]
+            decoded_json = json.loads(disk_info_json) 
+
+            for channel in range(len(decoded_json['Controller']['Channel'])) :
+                if 'HardDrive' in decoded_json['Controller']['Channel'][channel]:
+                    hd_disks_num = len(decoded_json['Controller']['Channel'][channel]['HardDrive'])
+                    if hd_disks_num > 0:
+                        for hd_disk in range(hd_disks_num):
+                            rc_lsm_disks.append(Arcconf._arcconf_disk_to_lsm_disk(decoded_json['Controller']['Channel'][channel]['HardDrive'][hd_disk], sys_id, cntrl_num, flag_free=False))            
+
+
+        return search_property(rc_lsm_disks, search_key, search_value)
+
+    @_handle_errors
```

Comment:
I would remove anything that returns "None" like this and remove support in the 'Capabilities' method for them. It's incorrect to label them as supported if they aren't yet (support can always be added later).